File: Leetcoding-Actions/LC-Monthly-Challenges/2020-11/08-Two-Sum-Less-Than-K.java
class Solution {
    public int twoSumLessThanK(int[] A, int K) {
        /*
        T: O(n) and S: O(1)
        */
        Arrays.sort(A);
        int i = 0;
        int j = A.length - 1;
        int maxSum = -1;
        while (i < j) {
            int sum = A[i] + A[j];
            if (sum < K) {
                maxSum = Math.max(maxSum, sum);
                i++;
            } else {
                j--;
            }
        }
        return maxSum;
    }
}

File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-11/08-Two-Sum-Less-Than-K.py
class Solution:
    def twoSumLessThanK(self, A: List[int], K: int) -> int:
        '''
        T: O(n) and S: O(1)
        '''
        A.sort()
        maxSum = -1
        i, j = 0, len(A) - 1
        while i < j:
            add = A[i] + A[j]
            if add < K:
                maxSum = max(maxSum, add)
                i += 1
            else:
                j -= 1
        return maxSum
